consensus/istanbul/core/core.go

```diff
@@ -75,8 +76,10 @@ type core struct {
 	valSet     istanbul.ValidatorSet
 	validateFn func([]byte, []byte) (common.Address, error)
 
-	backlogs   map[istanbul.Validator]*prque.Prque
-	backlogsMu *sync.Mutex
+	backlogBySeq      map[uint64]*prque.Prque
+	backlogCountByVal map[common.Address]int
```

Comment:
Should this be an lru cache? Not sure if this is a practical concern, but otherwise we'll keep keys of old addresses around forever even if they have been rotated

Reply:
Should be this be `uint` instead of `int`?

Reply:
> Should this be an lru cache? Not sure if this is a practical concern, but otherwise we'll keep keys of old addresses around forever even if they have been rotated

Won't we clear them out when we drain the backlog?

Reply:
yeah, I think if we just remove entries when the value falls to zero should be ok.  Then the map is at most 10k entries worst case. (and in practice, way smaller)

Reply:
Basically was looking for https://github.com/celo-org/celo-blockchain/pull/517/files/6f0aed30022a89bb06009932c12969fc7a521cd8#r348840207